src/state_machine/coordinator/mod.rs

```diff
@@ -0,0 +1,93 @@
+use p256k1::point::Point;
+
+use crate::{
+    common::MerkleRoot, errors::AggregatorError, net::Packet, state_machine::OperationResult,
+};
+
+#[derive(Debug, PartialEq)]
+/// Coordinator states
+pub enum State {
+    /// The coordinator is idle
+    Idle,
+    /// The coordinator is distributing public shares
+    DkgPublicDistribute,
+    /// The coordinator is gathering public shares
+    DkgPublicGather,
+    /// The coordinator is distributing private shares
+    DkgPrivateDistribute,
+    /// The coordinator is gathering DKG End messages
+    DkgEndGather,
+    /// The coordinator is requesting nonces
+    NonceRequest(bool, Option<MerkleRoot>),
+    /// The coordinator is gathering nonces
+    NonceGather(bool, Option<MerkleRoot>),
+    /// The coordinator is requesting signature shares
+    SigShareRequest(bool, Option<MerkleRoot>),
+    /// The coordinator is gathering signature shares
+    SigShareGather(bool, Option<MerkleRoot>),
+}
+
+#[derive(thiserror::Error, Debug)]
+/// The error type for the coordinator
+pub enum Error {
+    /// A bad state change was made
+    #[error("Bad State Change: {0}")]
+    BadStateChange(String),
+    /// A bad dkg_id in received message
+    #[error("Bad dkg_id: got {0} expected {1}")]
+    BadDkgId(u64, u64),
+    /// A bad sign_id in received message
+    #[error("Bad sign_id: got {0} expected {1}")]
+    BadSignId(u64, u64),
+    /// A bad sign_iter_id in received message
+    #[error("Bad sign_iter_id: got {0} expected {1}")]
+    BadSignIterId(u64, u64),
+    /// SignatureAggregator error
+    #[error("Aggregator: {0}")]
+    Aggregator(AggregatorError),
+    /// Schnorr proof failed to verify
+    #[error("Schnorr Proof failed to verify")]
+    SchnorrProofFailed,
+    /// No aggregate public key set
+    #[error("No aggregate public key set")]
+    MissingAggregatePublicKey,
+    /// No schnorr proof set
+    #[error("No schnorr proof set")]
+    MissingSchnorrProof,
+    /// No signature set
+    #[error("No signature set")]
+    MissingSignature,
+}
+
+impl From<AggregatorError> for Error {
+    fn from(err: AggregatorError) -> Self {
+        Error::Aggregator(err)
+    }
+}
+
+/// Coordinatable trait for handling the coordination of DKG and sign messages
+pub trait Coordinatable {
+    /// Process inbound messages
+    fn process_inbound_messages(
+        &mut self,
+        packets: Vec<Packet>,
```

Comment:
This is really just a side note...but could you as part of this pr change this to packets: &[Packet]?

It will remove some unnecsesary clones on the signer side as I believe we already take a reference internal to process_inbound_messages which implies we don't need to own this data.